meta-resin-common/recipes-containers/docker-disk/files/entry.sh

```diff
@@ -10,7 +10,7 @@ PARTITION_SIZE=${PARTITION_SIZE:=1024}
 
 # Create a BTRFS disk image
 dd if=/dev/zero of=/export/data_disk.img bs=1M count=$PARTITION_SIZE
-mkfs.btrfs /export/data_disk.img
+mkfs.btrfs --mixed --metadata=single /export/data_disk.img
```

Comment:
@lorenzo-stoakes single copy metadata increase the possiblity of corruption if the underlying storage is unreliable. i.e if the SD card flips bits. `mkfs.btrfs` will default to `single` if the underlying storage is not rotational (i.e SSD) so it seems like a misconfiguration caused due to the yocto builds being run on rotational storage.

The mixed mode is a recommendation of btrfs itself for small devices. It is mentioned in the man page of mkfs.btrfs. I didn't do any testing, I trust that they know what they are talking about. 

> The recommended size for the mixed mode is for filesystems less than 1GiB. The soft recommendation is to use it for filesystems smaller than 5GiB. The mixed mode may lead to degraded performance on larger filesystems, but is otherwise usable, even on multiple devices.

Regarding the emergency reserve, we didn't change any configuration there nor I see anyway of configuring it. According to docs, it should be allocated automatically.